lib/ip.go

```diff
@@ -157,22 +157,49 @@ func RandIP4ListWrite(n int, noBogon bool) {
 // `startIP` and `endIP` are the start & end IPs to generate IPs between.
 // `n` is the number of IPs to generate.
 // `noBogon`, if true, will ensure that none of the generated IPs are bogons.
+// `unique`, if true, will ensure every IP generated is unique.
 func RandIP4RangeListWrite(
 	startIP string,
 	endIP string,
 	n int,
 	noBogon bool,
+	unique bool,
 ) error {
 	ipRange, err := NewIP4Range(startIP, endIP)
 	if err != nil {
 		return err
 	}
-	for i := 0; i < n; i++ {
-		ip, err := RandIP4Range(ipRange, noBogon)
-		if err != nil {
-			return err
+	if unique {
+		// ensure range is larger than number of IPs to generate.
+		if uint32(ipRange.endIP-ipRange.startIP+1) < uint32(n) {
+			return errors.New("range is too small for unique IPs")
+		}
+
+		uniqueIP := make(map[uint32]net.IP)
```

Comment:
Notice that we don't actually need to use the value of this map, only the key. So we can make it an empty structure type:

```go
make(map[uint32]struct{})
```

Then when we assign something to a key, just assign `struct{}{}`:

```go
uniqueIP[ipInt] = struct{}{}
```

Then, immediately after this, we can print the IP in string form, instead of saving it in the map and printing it later as is currently being done.

Please do this for v6 as well.